js/src/logger.ts

```diff
@@ -4045,7 +4045,12 @@ function startSpanAndIsLogger<IsAsyncFlush extends boolean = true>(
 ): { span: Span; isSyncFlushLogger: boolean } {
   const state = args?.state ?? _globalState;
 
-  const parentStr = args?.parent ?? state.currentParent.getStore();
+  // Precedence: current span > current parent > logger
```

Comment:
@manugoyal can you sanity check this reasoning

Reply:
Would it make sense to implement this whole block of logic in `getSpanParentObject`? Currently that thing returns a `Span` object from which you call `startSpan`, but it could also return a `SpanComponentsV3` object from which you initialize the span directly.

I feel like `getSpanParentObject` ought to consider the parent string, and spreading out the selection between this function and that seems confusing.

That being said, I do think the precedence ordering is sane.